addons/mail/models/mail_thread.py

```diff
@@ -454,6 +454,10 @@ def _check_can_update_message_content(self, messages):
         if any(message.message_type != 'comment' for message in messages):
             raise exceptions.UserError(_("Only messages type comment can have their content updated"))
 
+    @api.model
+    def _get_from_request_or_raise(self, request, thread_id):
+        return self.search([("id", "=", thread_id)])
```

Comment:
why not to keep `browse(thread_id).exists()`?

Reply:
I tried to reduce number of queries in theory. `exists` always makes an query, whereas `search` is supposedly not making extra query if values are already in cache since https://github.com/odoo/odoo/issues/112126